### PR TITLE
Deprecation removals

### LINKED ISSUE #123: Enact actions in deprecation warnings of 0.3.0
Enact whatever actions are listed in the deprecation warnings in the 0.3.0 release notes.

### PR DESCRIPTION
**Pull request by DanPuzzuoli**
### Summary

Closes #123 

### Details and comments

As per the [deprecation notes of 0.3.0](https://qiskit.org/documentation/dynamics/release_notes.html), this PR removes:
- Support for passing `carriers` as a list in `InstructionToSignals`.
- The `Solver.signals` property, and the `Solver` init kwargs `hamiltonian_signals` and `lindblad_signals`.
    - Associated with this, the `Solver.solve` method has been updated to always reset the `Solver.model` signals to `None`.
- The `Solver.copy` method has been removed.

Tests that checked for the deprecation warnings have been removed. I've added a test that verifies that the `Solver.model.signals` property is always returned to `None` (or `(None, None)` for a Lindblad model) after calling `Solver.solve`.

### CODE REVIEW

File: releasenotes/notes/deprecation-removals-3d445f09f3fa0044.yaml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+---
+upgrade:
+  - |
+    Passing the ``carriers`` argument to ``InstructionToSignals`` as a ``list`` is no longer
+    supported.
+  - The ``Solver.signals`` property, and :class:`.Solver` init arguments ``hamiltonian_signals``
```
Comment (brosand) [inline releasenotes/notes/deprecation-removals-3d445f09f3fa0044.yaml:6]: this line is a bit confusing to read. Maybe just split into two lines

**Review by brosand — COMMENTED** [review]
LGTM, just left one small grammar piece, in the future I could just fix

File: releasenotes/notes/deprecation-removals-3d445f09f3fa0044.yaml (hunk 1 above)
Comment (DanPuzzuoli) [reply releasenotes/notes/deprecation-removals-3d445f09f3fa0044.yaml]: Broke it into separate lines. Also I realized that I didn't use proper formatting for the bullet points. Fixed each one to have format:

```
- |
   This is the bullet text.
```

**Review by brosand — APPROVED** [review]
LGTM

**Pull request merged**